Userland/Libraries/LibJS/Runtime/Date.cpp

```diff
@@ -483,8 +483,7 @@ double make_day(double year, double month, double date)
     if (!AK::is_within_range<int>(ym) || !AK::is_within_range<int>(mn + 1))
         return NAN;
 
-    // FIXME: We are avoiding AK::years_to_days_since_epoch here because it is implemented by looping over
-    //        the range [1970, ym), which will spin for any time value with an extremely large year.
+    // We cannot use AK::{years_to_,}days_since_epoch, because it might change the values slightly.
```

Comment:
Let's leave the FIXME in this comment. I remember when I wrote this that it felt like a shame that we couldn't use `years_to_days_since_epoch`. If the values are different, there's probably something worth looking into in LibJS.

Reply:
I didn't try replacing it, so I'm not entirely certain that the values would differ.

However, it seems that all of the rounding and overflow stuff is hardcoded in the spec, so replacing it by a call to `{years_to_,}days_since_epoch` means that it's harder to verify that we satisfy the spec.

If you want to keep the FIXME in, how do you suggest phrasing it? In particular, I don't think there is anything left to fix here: We adhere to the spec. The comment explains why we can't do it the "better" way.

Reply:
Essentially step 8 would here would become:
```js
    // 8. Find a finite time value t such that YearFromTime(t) is ym and MonthFromTime(t) is mn and DateFromTime(t) is 1𝔽; but if this is not possible (because some argument is out of range), return NaN.
    if (!AK::is_within_range<int>(ym) || !AK::is_within_range<int>(mn + 1))
        return NAN;

    auto t = (years_to_days_since_epoch(static_cast<int>(ym)) + day_of_year(static_cast<int>(ym), static_cast<int>(mn) + 1, 1)) * ms_per_day;
    VERIFY(year_from_time(t) == ym);
    VERIFY(month_from_time(t) == mn);
    VERIFY(date_from_time(t) == 1);
```

I think this would be better, because what I originally wrote is mixing `Date.prototype` math with `AK::Time` math, whereas this approach sticks with `AK::Time` math. It just wasn't feasible with the slow `for`-loops before.

Sidenote, when the spec says "Find a finite value" or something similar without actually telling you *how* to do it, we're in implementation-defined territory. We can/should verify that any invariant holds true (such as the `VERIFY`s written above), but beyond that, we're simultaneously on our own and can do we want (within reason).

Reply:
(To be clear, I'm not asking you to be responsible for that change here, just to leave the FIXME so we can do it another time)

Reply:
Took a hot minute to remember this, but basically the FIXME is reverting 032664332be6e1d42567165e9ffb623faf2b9a10.

Reply:
That looks good! I'll use that, thank you!